ClinicalTrial_SE_curation.Rmd

```diff
@@ -0,0 +1,320 @@
+---
+title: "Curation of Clinical trial Sumamrized Experiment objects"
+author: "Nikta Feizi, Christopher Eeles, Sisira Nair"
+date: "`r format(Sys.time(), '%a/%d/%b')`"
+output: html_document
+---
+#Please note:
+Parameters of "run_all" function are data specific and must be defined by looking at the metadata outside of the function. 
+Run "assay_data <- as.data.frame(exprs(rma.norm)) " to find "delim" which is the delimiter that separates the GSM IDs from the assay_data column names.   
+Pheno_data  column names must be used to assign column names to pre-defined columns (in "prior" vector). 
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+
+```{r}
+# configure download options for this session
+options(encoding = "UTF-8" , timeout = 1000)
+work_dir = setwd(paste0(getwd()))
+```
+
+```{r}
+# fetch Gencode v33 annotations from BHKLAB-Pachyderm/Annotations
+gencode_url <- "https://github.com/BHKLAB-Pachyderm/Annotations/raw/master/Gencode.v33.annotation.RData"
+gencode_file <- file.path(work_dir, "Ensembl.v99.annotation.RData")
+download.file(gencode_url, destfile=gencode_file)
+load("Ensembl.v99.annotation.RData")
+```
+
+
+```{r}
+#required libraries
+library(stringr)
+library(GEOquery)
+library(SummarizedExperiment)
+library(qs)
+library(affy)
+```
+
+```{r}
+geo_rma<- function (raw_tar, geo_path , brain_path){
+  
+  # Download CEL files
+  cel.file <- paste(work_dir, raw_tar, sep="/")
```

Comment:
Best Practices:

TL;DR - When modifying global options inside a function, ALWAYS add this before changing them with `options`
```
opts <- options()
on.exit(options(opts))
```

Setting options using the `options` function modifies a users global environment. Because functions in R should be totally self contained, and NOT change the global state of a users R environment after running, it is necessary to reset them when the function execution ends.

This can be easily achieved with the `on.exit` function, which registers an R expression to be run when the current function finishes executing.

First, you need to capture the current user options before modifying them:
`opts <- options()`

Then you register an expression to reset the user options when exiting a function:
`on.exit(options(opts))`

Then you can modify whichever options you need to for your function to work correctly.
```options( ... )```

Bugs caused when functions modify the global state of an R session are some of the hardest to figure out, and can end up wasting HOURS of debugging time.

Reply:
@NiktaFeizi

Reply:
Done

Reply:
I don't see these changes? Did you maybe forget to push them to the code_review branch?

Reply:
This was not quite correct. You need to capture the options inside the function before setting on.exit. Otherwise you have a function that requires a variable defined outside of its scope.

I have corrected it so we can get this PR completed.